[Docs] Autocomplete cookbook (#231)
Sylius/Stack: docs/cookbook/admin_panel/autocompletes.md

# Using autocompletes

[SyliusBootstrapAdminUi package](../../bootstrap-admin-ui/getting-started.md) uses [Symfony UX](https://ux.symfony.com/) under the hood.
Thus, UX autocomplete is already setup and configured in your admin panel.

## Configure entity autocomplete route for admin section

{% tabs %}
{% tab title="YAML" %}
{% code lineNumbers="true" %}
```yaml
# config/routes/ux_autocomplete.yaml
# ...
ux_entity_autocomplete_admin:
    path: '/admin/autocomplete/{alias}'
    controller: 'ux.autocomplete.entity_autocomplete_controller'
```
{% endcode %}
{% endtab %}

{% tab title="PHP" %}
{% code lineNumbers="true" %}
```php
<?php

> // config/routes/ux_autocomplete.php

Reviewer: ```suggestion
{% code lineNumbers="true" title="config/routes/ux_autocomplete.php" %}
```php
<?php
```

{
    public function configureOptions(OptionsResolver $resolver): void
    {
        $resolver->setDefaults([
            'class' => Speaker::class,
            'choice_label' => 'fullName',
        ]);
    }

    public function getParent(): string
    {
        return BaseEntityAutocompleteType::class;
    }
}
```

Then you need to create the grid filter.

```php
<?php

declare(strict_types=1);

namespace App\Grid\Filter;

use App\Form\SpeakerAutocompleteType;
use Sylius\Component\Grid\Data\DataSourceInterface;
use Sylius\Component\Grid\Filter\EntityFilter;
use Sylius\Component\Grid\Filtering\ConfigurableFilterInterface;

final class SpeakerFilter implements ConfigurableFilterInterface
{
    public function __construct(
        private readonly EntityFilter $entityFilter,
    ) {
    }

    public function apply(DataSourceInterface $dataSource, string $name, mixed $data, array $options): void
    {
        $this->entityFilter->apply($dataSource, $name, $data, $options);
    }

    public static function getFormType(): string
    {
        return SpeakerAutocompleteType::class;
    }

    public static function getType(): string
    {

>         return self::class; // it will allow to use FQCN instead of a string key.

Reviewer: ```suggestion
        return self::class; // this will allow us to use FQCN instead of a string key.
```

    }
}
```

We also need to configure the Twig template for our new grid filter.

```yaml

> # config/packages/sylius_grid.yaml

Reviewer: Place inside a code tag 
{% code lineNumbers="true" title="config/packages/sylius_grid.yaml" %}
```suggestion
{% code lineNumbers="true" title="config/packages/sylius_grid.yaml" %}
``yaml
```

sylius_grid:
    # ...
    templates:
        filter:
            'App\Grid\Filter\SpeakerFilter': '@SyliusBootstrapAdminUi/shared/grid/filter/entity.html.twig'

> ```

Reviewer: {% endcode %}

Now our new grid filtered is configured, we can use it in a grid.

> 

Reviewer: ```suggestion
{% code lineNumbers="true" title="src/Grid/TalkGrid.php" %}
```
